src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/AbstractOverviewIndexWriter.java

```diff
@@ -62,35 +62,35 @@ public AbstractOverviewIndexWriter(HtmlConfiguration configuration,
      * summary at the top of the page and generate a link to the description,
      * which is added at the end of this page.
      *
-     * @param main the documentation tree to which the overview header will be added
+     * @param target the documentation to which the overview header will be added
      */
-    protected void addOverviewHeader(Content main) {
-        addConfigurationTitle(main);
-        addOverviewComment(main);
-        addOverviewTags(main);
+    protected void addOverviewHeader(Content target) {
+        addConfigurationTitle(target);
+        addOverviewComment(target);
+        addOverviewTags(target);
```

Comment:
I could revert it if you'd like.

While refactoring, I followed a principle that if a method is specific, then its arguments should hint on that specificity. Conversely, if a method could be used in more than a single context, then its arguments should have general names. For example, the below methods should only be passed the HEAD HTML element, so their respective `HtmlTree` arguments are called "head" rather than "target":

  * jdk.javadoc.internal.doclets.formats.html.markup.Head#addStylesheets
  * jdk.javadoc.internal.doclets.formats.html.markup.Head#addStylesheet
  * jdk.javadoc.internal.doclets.formats.html.markup.Head#addScripts

So it's not that I don't agree with you, it's that sometimes I miss stuff. Which is not that weird given that plethora of tiny, incohesive methods that plague writers.

Since we're here, the way builders and writers work together to assemble an HTML tree is strange [^1]. Although I think that it is a result of evolution, we have to fix it. I'm not ready to have this conversation right now, as I think this PR is already doing way too much beyond its original intent.

[^1]: I hope to investigate alternatives in https://bugs.openjdk.java.net/browse/JDK-8283576.